src/intersection-observer-polyfill.js

```diff
@@ -233,60 +259,66 @@ export class IntersectionObserverPolyfill {
-        elementRect, ownerRect, hostViewport, opt_iframe);
-    if (changeEntry) {
-      this.callback_(changeEntry);
+    if (changes.length) {
+      this.callback_(changes);
     }
   }
 
   /**
-   * Return a change entry for that should be compatible with
+   * Return a change entry for one element that should be compatible with
    * IntersectionObserverEntry if it's valid with current config.
    * When the new intersection ratio doesn't cross one of a threshold value,
    * the function will return null.
    *
-   * Mutates passed in rootBounds to have x and y according to spec.
-   *
-   * @param {!./layout-rect.LayoutRectDef} element element's rect
-   * @param {?./layout-rect.LayoutRectDef} owner element's owner rect
+   * @param {!observeEntryDef} entry
    * @param {!./layout-rect.LayoutRectDef} hostViewport hostViewport's rect
    * @param {./layout-rect.LayoutRectDef=} opt_iframe. iframe container rect
    * @return {?IntersectionObserverEntry} A valid change entry or null if ratio
-   * does not fill in a new threshold bucket.
    * @private
    */
-  getValidIntersectionChangeEntry_(element, owner, hostViewport, opt_iframe) {
+  getValidIntersectionChangeEntry_(entry, hostViewport, opt_iframe) {
+    const element = entry.element;
+
+    // Normalize container LayoutRect to be relative to page
+    let elementRect;
+    let ownerRect = null;
+
+    // If opt_iframe is provided, all LayoutRect has position relative to
+    // the iframe.
+    // If opt_iframe is not provided, all LayoutRect has position relative to
+    // the host document.
+    elementRect = element.getLayoutBox();
+    const owner = element.getOwner();
+    ownerRect = owner && owner.getLayoutBox();
+
     // calculate intersectionRect. that the element intersects with hostViewport
     // and intersects with owner element and container iframe if exists.
     const intersectionRect =
-        rectIntersection(element, owner, hostViewport, opt_iframe) ||
+        rectIntersection(elementRect, ownerRect, hostViewport, opt_iframe) ||
         layoutRectLtwh(0, 0, 0, 0);
-
     // calculate ratio, call callback based on new ratio value.
-    const ratio = intersectionRatio(intersectionRect, element);
+    const ratio = intersectionRatio(intersectionRect, elementRect);
     const newThresholdSlot = getThresholdSlot(this.threshold_, ratio);
-    if (newThresholdSlot == this.prevThresholdSlot_) {
+
+    if (newThresholdSlot == entry.currentThresholdSlot) {
       return null;
     }
-    this.prevThresholdSlot_ = newThresholdSlot;
+    entry.currentThresholdSlot = newThresholdSlot;
 
     // To get same behavior as native IntersectionObserver set hostViewport null
     // if inside an iframe
-    return calculateChangeEntry(
-        element, (opt_iframe ? null : hostViewport), intersectionRect, ratio);
+    const changeEntry = calculateChangeEntry(elementRect,
+        (opt_iframe ? null : hostViewport), intersectionRect, ratio);
+    changeEntry.target = element;
```

Comment:
I get that, but we're the consumers and we're going to delete it before sending it to anyone else. I don't see why we need to set it to begin with.